### PR TITLE
Add telemetry to track reasons for disconnects

### LINKED ISSUE #132902: Add telemetry to track reasons for disconnects
Add telemetry for disconnects to help with debugging.

### PR DESCRIPTION
**Pull request by phylix**
Added telemetry to track reasons for disconnects to help with debugging.

This PR fixes #132902

### CODE REVIEW

**Comment by roblourens** [comment]
Is there some other context or discussion behind this?

File: src/vs/base/parts/ipc/common/ipc.net.ts [hunk 1]
```diff
@@ -737,6 +740,10 @@ export class PersistentProtocol implements IMessagePassingProtocol {
 		this._socketDisposables = dispose(this._socketDisposables);
 	}
 
+	logTelemetry(event: string, data: Object): void {
```
Comment (alexdima) [inline src/vs/base/parts/ipc/common/ipc.net.ts:743]: It doesn't look to me like there are any callers for this method

File: src/vs/platform/remote/common/remoteAgentConnection.ts [hunk 2]
```diff
@@ -556,13 +559,21 @@ abstract class PersistentConnection extends Disposable {
...
+		}));
+
 		PersistentConnection._instances.push(this);
 
 		if (PersistentConnection._permanentFailure) {
 			this._gotoPermanentFailure(PersistentConnection._permanentFailureMillisSinceLastIncomingData, PersistentConnection._permanentFailureAttempt, PersistentConnection._permanentFailureHandled);
 		}
 	}
 
+	public logTelemetryEvent(event: string, data: Object): void {
```
Comment (alexdima) [inline src/vs/platform/remote/common/remoteAgentConnection.ts:573]: Also, there are no callers for this method

**Review by alexdima — CHANGES_REQUESTED** [review]
I like the idea of logging the extension host exit reason, but there are some problems with the PR (some methods are not called), and I would expect to see the handling of the `process.send` event for the local extension host.

**Pull request closed without merging**